lib/policy_machine_storage_adapters/active_record/postgresql.rb

```diff
@@ -74,6 +74,40 @@ def self.operations_for_operation_sets(operation_set_ids, operation_names = nil)
         # ]
         connection.execute(sanitized_query)
       end
+
+      def self.accessible_objects_for_operations(user_id, operation_names, options)
+        field = options[:fields].first
+        filters = options.dig(:filters, :user_attributes) || {}
+
+        query = <<~SQL.squish
+          SELECT *
+          FROM pm_accessible_objects_for_operations(?, ?, ?, ?)
+        SQL
+
+        sanitized_query = sanitize_sql_for_assignment([
+          query,
+          user_id,
+          ActiveRecord.array_to_string(operation_names),
+          field,
+          JSON.dump(filters)
+        ])
+
+        # [
+        #   { 'unique_identifier' => 'op1', 'objects' => '{obj1,obj2,obj3}' },
```

Comment:
Yes, this is the format that postgres represents arrays and JSON https://www.postgresql.org/docs/current/arrays.html#ARRAYS-INPUT.

Incoming commit will use `PG::TextDecoder::Array` instead of my DIY version which uses C extensions.

Reply:
24b4a9d